prebuild/emscripten/post.js

```diff
@@ -0,0 +1,1010 @@
+    }
+    sig = sig.ptr;
+    return _emscripten_bind_Pairing_aggregate_6__P2_P1(self, pk, sig, _msg, msg_len, _aug, aug_len);
+  }
+};;
+
+Pairing.prototype['mul_n_aggregate'] = Pairing.prototype.mul_n_aggregate = /** @suppress {undefinedVars, duplicate} @this{Object} */function(pk, sig, scalar, msg, aug) {
+  var self = this.ptr;
+  ensureCache.prepare();
+
+  const [_scalar, scalar_len] = ensureRef(scalar);
+  const [_msg, msg_len] = ensureRef(msg);
+  const [_aug, aug_len] = ensureRef(aug);
+
+  if (pk.constructor === P1_Affine) {
+    pk = pk.ptr;
+
+    if (sig.constructor !== P2_Affine) {
+      throw new Error("`pk` is P1_Affine, expected `sig` to be P2_Affine; received: " + sig.constructor.name);
+    }
+    sig = sig.ptr;
+    return _emscripten_bind_Pairing_mul_n_aggregate_8__P1_P2(self, pk, sig, _scalar, scalar_len, _msg, msg_len, _aug, aug_len);
+  } else {
+    pk = pk.ptr;
+
+    if (sig.constructor !== P1_Affine) {
+      throw new Error("`pk` is P2_Affine, expected `sig` to be P1_Affine; received: " + sig.constructor.name);
+    }
+    sig = sig.ptr;
+    return _emscripten_bind_Pairing_mul_n_aggregate_8__P2_P1(self, pk, sig, _scalar, scalar_len, _msg, msg_len, _aug, aug_len);
+  }
+};;
+
+Pairing.prototype['commit'] = Pairing.prototype.commit = /** @suppress {undefinedVars, duplicate} @this{Object} */function() {
+  var self = this.ptr;
+  _emscripten_bind_Pairing_commit_0(self);
+};;
+
+Pairing.prototype['merge'] = Pairing.prototype.merge = /** @suppress {undefinedVars, duplicate} @this{Object} */function(ctx) {
+  var self = this.ptr;
+  if (ctx && typeof ctx === 'object') ctx = ctx.ptr;
+  return _emscripten_bind_Pairing_merge_1(self, ctx);
+};;
+
+Pairing.prototype['finalverify'] = Pairing.prototype.finalverify = /** @suppress {undefinedVars, duplicate} @this{Object} */function(sig) {
+  var self = this.ptr;
+  if (sig && typeof sig === 'object') sig = sig.ptr;
+  return !!(_emscripten_bind_Pairing_finalverify_1(self, sig));
+};;
+
+Pairing.prototype['__destroy__'] = Pairing.prototype.__destroy__ = /** @suppress {undefinedVars, duplicate} @this{Object} */function() {
+  var self = this.ptr;
+  _emscripten_bind_Pairing___destroy___0(self);
+};
+
+// G1
+/** @suppress {undefinedVars, duplicate} @this{Object} */function G1() {
+  return wrapPointer(_emscripten_bind_G1_0(), P1);
+};;
+Module['G1'] = G1;
+
+// G2
+/** @suppress {undefinedVars, duplicate} @this{Object} */function G2() {
+  return wrapPointer(_emscripten_bind_G2_0(), P2);
+};;
+Module['G2'] = G2;
+
+function ensureRef(val) {
+  // TODO: something else (nullptr)?
+  if (typeof val === "undefined") {
+    return [ensureString(""), 0];
+  }
+
+  switch (val.constructor.name) {
+    case "String":
+      return [ensureString(val), val.length];
+    case "Buffer":
+      // NB: intentional fallthrough.
+    case "Uint8Array":
+      return [ensureUint8(val), val.length];
+    // TODO:
+    // case BigInt:
```

Comment:
This one would go something like this
```
var temp = [];
while (val != 0) {
    temp.push(Number(val & 255n));
    val >>= 8n;
}
return "whatevermagic that copies Uint8Array(temp) to wasm memory"
```
[As you can tell, I don't have a firm grasp on how things get to the wasm memory. I only know that it's separate, that Javascript can read and write it, while wasm can **not** read, let alone write, the Javascript memory. Right?]

Reply:
Thanks! This binary logic seems very on-brand :stuck_out_tongue_winking_eye: 

I don't think webpack will like the bigint literals, so maybe don't be surprised `BigInt(...)`s come around later.

Reply:
In essence the idea would be to fully harmonize with [blst.hpp.ts](https://github.com/supranational/blst/blob/master/bindings/node.js/blst.hpp.ts).